engineio/src/model.rs

```diff
@@ -0,0 +1,9 @@
+use crate::error::Result;
+use crate::Packet;
+
+pub trait Socket {
+    fn close(&self) -> Result<()>;
+    fn connect(&self) -> Result<()>;
+    fn emit(&self, packet: Packet) -> Result<()>;
+    fn is_connected(&self) -> Result<bool>;
```

Comment:
Nice! Everything with `&_` allows to just pass an `&Socket` to the callback!

Reply:
EngineIO doesn't pass the socket to engineio callbacks today, only socketio exposes a socket in the callbacks.